File: fragileRotatingBox.py
def fragileRotatingBox(boxWeakness, surfaceRoughness):
    bottom = ""
    top = ""
    right = ""
    left = ""

    bottom = boxWeakness[len(boxWeakness) - 1]
    top = boxWeakness[0][::-1]
    for i in range(len(boxWeakness)):
        e = boxWeakness[i]
        right += e[len(e) - 1]
        left += e[0]

    right = right[::-1]

    rotate = ""
    sides = [bottom, right, top, left]
    index = 0
    isGoing = True

    while isGoing:
        if len(rotate + sides[index]) <= len(surfaceRoughness):
            rotate += sides[index]
        else:
            for i in range(len(sides[index])):
                if(len(rotate + sides[index][i]) <= len(surfaceRoughness)):
                    rotate += sides[index][i]
                else:
                    isGoing = False
                    break

        index += 1
        if index == 4:
            index = 0

        pass
    ans = 0
    for i in range(len(rotate)):
        ans += int(rotate[i]) * int(surfaceRoughness[i])

    return ans

File: fragileRotatingBox.js
function fragileRotatingBox(boxWeakness, surfaceRoughness) {
    let bottom = boxWeakness[boxWeakness.length-1];
    let right = '', left = '', top = '';
    
    boxWeakness.map(e=>{
        right += e[e.length-1];
    })
    
    right = right.split('').reverse().join('');
    top = boxWeakness[0].split('').reverse().join('');
    boxWeakness.map(e=>{
        left += e[0];
    })
    
    
    let ans = 0;
    let sides = [bottom, right, top, left];
    let index = 0;
    let rotate = '';
    let isGoing = true;
    
    while(isGoing){
        if((rotate + sides[index]).length <= surfaceRoughness.length){
            rotate += sides[index];
        }else{
            for(let i = 0;i< sides[index].length;i++){
                if((rotate + sides[index][i]).length <= surfaceRoughness.length){
                    rotate += sides[index][i];
                }else{
                    isGoing = false;
                    break;
                }
            }
        }
        
        index++;
        if(index == 4) index = 0;
    }
    
    for(let i = 0;i< rotate.length;i++){
        ans += Number(rotate[i]) * Number(surfaceRoughness[i]);
    }
    
    return ans;
}
